openllm-core/src/openllm_core/_configuration.py

```diff
@@ -1356,6 +1356,24 @@ def model_dump(self, flatten: bool = False, **_: t.Any) -> DictStrAny:
   def model_dump_json(self, **kwargs: t.Any) -> bytes:
     return orjson.dumps(self.model_dump(**kwargs))
 
+  def to_openai_config(self, new_config: t.Any) -> t.Any:
+    default_config = bentoml_cattr.unstructure(self)
+    generation_config = bentoml_cattr.unstructure(self.generation_config)
+    sampling_config = bentoml_cattr.unstructure(self.sampling_config)
+    default_config.update(generation_config)
+    default_config.update(sampling_config)
```

Comment:
what is the reason for doing this? shouldn't `to_openai_config` just get all of the available options that OpenAI API accepts?

All of the currently attributes for LLMConfig can still be accessed within the service definition.